Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -406,6 +407,11 @@ void Renderer::CheckForConfigChanges()
 
   UpdateActiveConfig();
 
+  if (g_ActiveConfig.bFreeLook)
+  {
+    g_freelook_camera.SetControlType(g_ActiveConfig.iFreelookControlType);
```

Comment:
Does this function _have_ to run every frame? Could you save the old freelook control type/freelook enable and only run it when that changes? e.g.
```cpp
const bool old_freelook = g_ActiveConfig.bFreeLook;
const int old_freelook_control_type = g_ActiveConfig.iFreelookControlType;

if (old_freelook != g_ActiveConfig.bFreeLook || old_freelook_control_type != g_ActiveConfig.iFreelookControlType)
{
  g_freelook_camera.SetControlType(g_ActiveConfig.iFreelookControlType);
}
```

Reply:
It doesn't _have_ to but what's the point of all this checking?  It doesn't really matter that function is called so often because the actual function already bails if there are no changes.  The whole point is to just get the new value from configuration.

Reply:
Oh, I didn't see the check here: https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoCommon/FreeLookCamera.cpp#L179

I misremembered it unconditionally reallocating it. In that case, it's fine as-is.